compiler/qsc_partial_eval/src/tests/branching.rs

```diff
@@ -2311,3 +2311,95 @@ fn if_expression_with_dynamic_operand_from_hybrid_doubles_array() {
                 Jump(6)"#]],
     );
 }
+
```

Comment:
We didn't have any "positive" tests for this, did we?

Reply:
We did not, because it turns out we never properly supported it. That's why it makes sense to reject it at compile time.